components/RestPasswordForm/ResetPasswordForm.test.tsx

```diff
@@ -0,0 +1,116 @@
+import { fireEvent, render, screen, waitFor } from '@testing-library/react';
+import { useRouter } from 'next/navigation';
+import { toast } from 'react-hot-toast';
+import ResetPasswordForm from './ResetPasswordForm';
+import { resetPassword } from '@/api/apiFunctions';
+import Alert from '@/components/AlertNotification/AlertNotification';
+import { AlertVariants } from '@/components/AlertNotification/Alerts.enum';
+import { useAuthContext } from '@/context/AuthContextProvider';
+
+jest.mock('react-hot-toast');
+jest.mock('next/navigation');
+jest.mock('@/context/AuthContextProvider', () => ({
+  useAuthContext: jest.fn(),
+}));
+jest.mock('@/api/apiFunctions');
+
+describe('ResetPassword', () => {
+  let mockRouter: { push: jest.Mock };
+  let mockToast: jest.Mock;
+  const mockLogoutAccount = jest.fn();
+
+  beforeEach(() => {
+    mockRouter = { push: jest.fn() };
+    mockToast = jest.fn();
+
+    (useAuthContext as jest.Mock).mockReturnValue({
+      logoutAccount: mockLogoutAccount,
+      isLoggedIn: true,
+    });
+    (useRouter as jest.Mock).mockReturnValue(mockRouter);
+    (toast.custom as jest.Mock).mockImplementation(mockToast);
+  });
+
+  afterEach(() => {
+    jest.clearAllMocks();
+  });
+
+  const fillForm = () => {
+    fireEvent.change(screen.getByTestId('old-password'), {
+      target: { value: 'oldPassword123' },
+    });
+    fireEvent.change(screen.getByTestId('new-password'), {
+      target: { value: 'newPassword123' },
+    });
+    fireEvent.click(screen.getByTestId('update-password-button'));
+  };
+
+  it('should render the reset password form with the update button disabled', async () => {
+    render(<ResetPasswordForm />);
+
+    expect(screen.getByTestId('old-password')).toBeInTheDocument();
+    expect(screen.getByTestId('new-password')).toBeInTheDocument();
+    expect(screen.getByTestId('update-password-button')).toBeDisabled();
```

Comment:
Should we be testing the values to be empty here as well since we're testing if the update-password-button is disabled?